### ISSUE TITLE
Unable to build project with CI when FMOD plugin is installed

### ISSUE THREAD

**Issue by GabLeRoux** [alessandrofama/fmod-for-godot#23]
### FMOD for Godot version

FMOD 2.02.17 for Godot 4.1.1 stable v1.0-alpha-4

### System information

ubuntu 20.04 (docker)

### Issue description

I've created the following blank Godot 4 project with a very basic bank (not even used in Godot project). I would like to build the project from CI/CD using the great [abarichello/godot-ci](https://github.com/abarichello/godot-ci) project.

There are several script errors and missing resources when running the project on the `fmod` branch. Most of these errors are related to missing types or identifiers in the `fmod_runtime_manager.gd` script, and missing editor-specific import metadata for some textures. Additionally, there's a fatal error which leads to a program crash.

<details>
<summary>See complete log</summary>

```
Godot Engine v4.1.2.stable.official.399c9dc39 - https://godotengine.org/
TextServer: Added interface "Dummy"
TextServer: Added interface "ICU / HarfBuzz / Graphite (Built-in)"
Using "default" pen tablet driver...
JoypadLinux: udev enabled and loaded successfully.
 
TextServer: Primary interface set to: "ICU / HarfBuzz / Graphite (Built-in)".
CORE API HASH: 206002781
EDITOR API HASH: 121686721
Regenerating editor help cache
Found Android plugin FmodAndroidPlugin
Class 'EditorPropertyNameProcessor' is not exposed, skipping.
Class 'GDScriptEditorTranslationParserPlugin' is not exposed, skipping.
Class 'GDScriptNativeClass' is not exposed, skipping.
Class 'GLTFDocumentExtensionPhysics' is not exposed, skipping.
Class 'GLTFDocumentExtensionTextureWebP' is not exposed, skipping.
Class 'GodotPhysicsServer2D' is not exposed, skipping.
Class 'GodotPhysicsServer3D' is not exposed, skipping.
Class 'IPUnix' is not exposed, skipping.
Class 'MovieWriterMJPEG' is not exposed, skipping.
Class 'MovieWriterPNGWAV' is not exposed, skipping.
Class 'ResourceFormatImporterSaver' is not exposed, skipping.
Class 'ResourceImporterMP3' is not exposed, skipping.
Class 'ResourceImporterOggVorbis' is not exposed, skipping.
Class 'SceneCacheInterface' is not exposed, skipping.
Class 'SceneRPCInterface' is not exposed, skipping.
Class 'SceneReplicationInterface' is not exposed, skipping.
WARNING: Custom cursor shape not supported by this display server.
     at: cursor_set_custom_image (servers/display_server.cpp:480)
SCRIPT ERROR: Parse Error: Could not find type "StudioSystem" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:3)
SCRIPT ERROR: Parse Error: Could not find type "EventAsset" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:36)
SCRIPT ERROR: Parse Error: Could not find type "EventDescription" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:36)
SCRIPT ERROR: Parse Error: Could not find type "EventDescription" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:42)
SCRIPT ERROR: Parse Error: Could not find type "EventDescription" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:48)
SCRIPT ERROR: Parse Error: Could not find type "EventAsset" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:54)
SCRIPT ERROR: Parse Error: Could not find type "EventInstance" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:54)
SCRIPT ERROR: Parse Error: Could not find type "EventInstance" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:63)
SCRIPT ERROR: Parse Error: Could not find type "EventInstance" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:71)
SCRIPT ERROR: Parse Error: Could not find type "EventAsset" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:79)
SCRIPT ERROR: Parse Error: Could not find type "EventAsset" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:101)
SCRIPT ERROR: Parse Error: Could not find type "EventInstance" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:126)
SCRIPT ERROR: Parse Error: Could not find type "EventInstance" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:143)
SCRIPT ERROR: Parse Error: Could not find type "EventInstance" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:8)
SCRIPT ERROR: Parse Error: Identifier "FMODStudioModule" not declared in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:17)
SCRIPT ERROR: Parse Error: Identifier "FMODStudioModule" not declared in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:18)
SCRIPT ERROR: Parse Error: Identifier "FMODDebugMonitor" not declared in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:20)
SCRIPT ERROR: Parse Error: Identifier "FMODStudioModule" not declared in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:29)
SCRIPT ERROR: Parse Error: Could not resolve member "studio_system".
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:33)
SCRIPT ERROR: Parse Error: Could not find type "EventDescription" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:37)
SCRIPT ERROR: Parse Error: Could not find type "EventDescription" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:44)
SCRIPT ERROR: Parse Error: Could not find type "EventDescription" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:49)
SCRIPT ERROR: Parse Error: Could not find type "EventDescription" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:55)
SCRIPT ERROR: Parse Error: Could not find type "EventInstance" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:58)
SCRIPT ERROR: Parse Error: Could not find type "EventDescription" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:64)
SCRIPT ERROR: Parse Error: Could not find type "EventInstance" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:66)
SCRIPT ERROR: Parse Error: Could not find type "EventDescription" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:72)
SCRIPT ERROR: Parse Error: Could not find type "EventInstance" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:74)
SCRIPT ERROR: Parse Error: Identifier "FMOD_3D_ATTRIBUTES" not declared in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:113)
SCRIPT ERROR: Parse Error: Identifier "RuntimeUtils" not declared in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:116)
SCRIPT ERROR: Parse Error: Identifier "RuntimeUtils" not declared in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:118)
SCRIPT ERROR: Parse Error: Identifier "RuntimeUtils" not declared in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:120)
SCRIPT ERROR: Parse Error: Identifier "FMOD_3D_ATTRIBUTES" not declared in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:135)
SCRIPT ERROR: Parse Error: Identifier "RuntimeUtils" not declared in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:136)
SCRIPT ERROR: Parse Error: Identifier "StudioEventEmitter3D" not declared in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:157)
SCRIPT ERROR: Parse Error: Could not find type "FMODStudioModule" in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:161)
SCRIPT ERROR: Parse Error: Identifier "FMODStudioModule" not declared in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:162)
SCRIPT ERROR: Parse Error: Identifier "FMODStudioModule" not declared in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:171)
SCRIPT ERROR: Parse Error: Identifier "FMOD_3D_ATTRIBUTES" not declared in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:179)
SCRIPT ERROR: Parse Error: Identifier "RuntimeUtils" not declared in the current scope.
          at: GDScript::reload (res://addons/FMOD/runtime/fmod_runtime_manager.gd:181)
ERROR: Script does not inherit from Node: res://addons/FMOD/runtime/fmod_runtime_manager.gd.
   at: _create_autoload (editor/editor_autoload_settings.cpp:423)
WARNING: Blend file import is enabled in the project settings, but no Blender path is configured in the editor settings. Blend files will not be imported.
     at: _editor_init (modules/gltf/register_types.cpp:73)
Loaded system CA certificates
EditorSettings: Save OK!
ERROR: Missing required editor-specific import metadata for a texture (please reimport it using the 'Import' tab): 'res://.godot/imported/bank_icon.svg-e88ff69e5163a57c4926ed9582cf5208.editor.meta'
   at: _load_editor_meta (editor/import/resource_importer_texture.cpp:420)
ERROR: Missing required editor-specific import metadata for a texture (please reimport it using the 'Import' tab): 'res://.godot/imported/bus_icon.svg-c3b0526565bf49c99329b343e8d00a40.editor.meta'
   at: _load_editor_meta (editor/import/resource_importer_texture.cpp:420)
ERROR: Missing required editor-specific import metadata for a texture (please reimport it using the 'Import' tab): 'res://.godot/imported/c_parameter_icon.svg-5b0fa045bdd1406175454131a0af0dff.editor.meta'
   at: _load_editor_meta (editor/import/resource_importer_texture.cpp:420)
ERROR: Missing required editor-specific import metadata for a texture (please reimport it using the 'Import' tab): 'res://.godot/imported/d_parameter_icon.svg-d7ec1115143890191303e60b552df542.editor.meta'
   at: _load_editor_meta (editor/import/resource_importer_texture.cpp:420)
ERROR: Missing required editor-specific import metadata for a texture (please reimport it using the 'Import' tab): 'res://.godot/imported/event_icon.svg-043c705723964fab9af2dd9922a92563.editor.meta'
   at: _load_editor_meta (editor/import/resource_importer_texture.cpp:420)
ERROR: Missing required editor-specific import metadata for a texture (please reimport it using the 'Import' tab): 'res://.godot/imported/fmod_icon.svg-f41d88c38b2d44ab7b714bc51d959622.editor.meta'
   at: _load_editor_meta (editor/import/resource_importer_texture.cpp:420)
ERROR: Missing required editor-specific import metadata for a texture (please reimport it using the 'Import' tab): 'res://.godot/imported/folder_closed.svg-ec1a61bfb462b3ab84fc9c69410d995e.editor.meta'
   at: _load_editor_meta (editor/import/resource_importer_texture.cpp:420)
ERROR: Missing required editor-specific import metadata for a texture (please reimport it using the 'Import' tab): 'res://.godot/imported/folder_opened.svg-40fae91eff73ee4d13ff7d45de2bc499.editor.meta'
   at: _load_editor_meta (editor/import/resource_importer_texture.cpp:420)
ERROR: Missing required editor-specific import metadata for a texture (please reimport it using the 'Import' tab): 'res://.godot/imported/l_parameter_icon.svg-b3150a4178950ddc36be6ab21128320e.editor.meta'
   at: _load_editor_meta (editor/import/resource_importer_texture.cpp:420)
ERROR: Missing required editor-specific import metadata for a texture (please reimport it using the 'Import' tab): 'res://.godot/imported/snapshot_icon.svg-936970fe2f6f021d3c4b6fe0302f3ba2.editor.meta'
   at: _load_editor_meta (editor/import/resource_importer_texture.cpp:420)
ERROR: Missing required editor-specific import metadata for a texture (please reimport it using the 'Import' tab): 'res://.godot/imported/vca_icon.svg-fc5bfda9f408aae7588fb7bb267e0bd0.editor.meta'
   at: _load_editor_meta (editor/import/resource_importer_texture.cpp:420)
ERROR: Unable to open file: res://.godot/imported/fmod_icon.svg-f41d88c38b2d44ab7b714bc51d959622.editor.ctex.
[FMOD] Saved Project Settings
   at: _load_data (scene/resources/texture.cpp:847)
ERROR: Failed loading resource: res://.godot/imported/fmod_icon.svg-f41d88c38b2d44ab7b714bc51d959622.editor.ctex. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Failed loading resource: res://addons/FMOD/editor/images/fmod_icon.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Error loading resource: 'res://addons/FMOD/editor/images/fmod_icon.svg'.
   at: load (core/core_bind.cpp:73)
ERROR: Unable to open file: res://.godot/imported/folder_closed.svg-ec1a61bfb462b3ab84fc9c69410d995e.editor.ctex.
   at: _load_data (scene/resources/texture.cpp:847)
ERROR: Failed loading resource: res://.godot/imported/folder_closed.svg-ec1a61bfb462b3ab84fc9c69410d995e.editor.ctex. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Failed loading resource: res://addons/FMOD/editor/images/folder_closed.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Error loading resource: 'res://addons/FMOD/editor/images/folder_closed.svg'.
   at: load (core/core_bind.cpp:73)
ERROR: Unable to open file: res://.godot/imported/folder_opened.svg-40fae91eff73ee4d13ff7d45de2bc499.editor.ctex.
   at: _load_data (scene/resources/texture.cpp:847)
ERROR: Failed loading resource: res://.godot/imported/folder_opened.svg-40fae91eff73ee4d13ff7d45de2bc499.editor.ctex. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Failed loading resource: res://addons/FMOD/editor/images/folder_opened.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Error loading resource: 'res://addons/FMOD/editor/images/folder_opened.svg'.
   at: load (core/core_bind.cpp:73)
ERROR: Unable to open file: res://.godot/imported/event_icon.svg-043c705723964fab9af2dd9922a92563.editor.ctex.
   at: _load_data (scene/resources/texture.cpp:847)
ERROR: Failed loading resource: res://.godot/imported/event_icon.svg-043c705723964fab9af2dd9922a92563.editor.ctex. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Failed loading resource: res://addons/FMOD/editor/images/event_icon.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Error loading resource: 'res://addons/FMOD/editor/images/event_icon.svg'.
   at: load (core/core_bind.cpp:73)
ERROR: Unable to open file: res://.godot/imported/bank_icon.svg-e88ff69e5163a57c4926ed9582cf5208.editor.ctex.
   at: _load_data (scene/resources/texture.cpp:847)
ERROR: Failed loading resource: res://.godot/imported/bank_icon.svg-e88ff69e5163a57c4926ed9582cf5208.editor.ctex. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Failed loading resource: res://addons/FMOD/editor/images/bank_icon.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Error loading resource: 'res://addons/FMOD/editor/images/bank_icon.svg'.
   at: load (core/core_bind.cpp:73)
ERROR: Unable to open file: res://.godot/imported/snapshot_icon.svg-936970fe2f6f021d3c4b6fe0302f3ba2.editor.ctex.
   at: _load_data (scene/resources/texture.cpp:847)
ERROR: Failed loading resource: res://.godot/imported/snapshot_icon.svg-936970fe2f6f021d3c4b6fe0302f3ba2.editor.ctex. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Failed loading resource: res://addons/FMOD/editor/images/snapshot_icon.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Error loading resource: 'res://addons/FMOD/editor/images/snapshot_icon.svg'.
   at: load (core/core_bind.cpp:73)
ERROR: Unable to open file: res://.godot/imported/bus_icon.svg-c3b0526565bf49c99329b343e8d00a40.editor.ctex.
   at: _load_data (scene/resources/texture.cpp:847)
ERROR: Failed loading resource: res://.godot/imported/bus_icon.svg-c3b0526565bf49c99329b343e8d00a40.editor.ctex. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Failed loading resource: res://addons/FMOD/editor/images/bus_icon.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Error loading resource: 'res://addons/FMOD/editor/images/bus_icon.svg'.
   at: load (core/core_bind.cpp:73)
ERROR: Unable to open file: res://.godot/imported/vca_icon.svg-fc5bfda9f408aae7588fb7bb267e0bd0.editor.ctex.
   at: _load_data (scene/resources/texture.cpp:847)
ERROR: Failed loading resource: res://.godot/imported/vca_icon.svg-fc5bfda9f408aae7588fb7bb267e0bd0.editor.ctex. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Failed loading resource: res://addons/FMOD/editor/images/vca_icon.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Error loading resource: 'res://addons/FMOD/editor/images/vca_icon.svg'.
   at: load (core/core_bind.cpp:73)
ERROR: Unable to open file: res://.godot/imported/c_parameter_icon.svg-5b0fa045bdd1406175454131a0af0dff.editor.ctex.
   at: _load_data (scene/resources/texture.cpp:847)
ERROR: Failed loading resource: res://.godot/imported/c_parameter_icon.svg-5b0fa045bdd1406175454131a0af0dff.editor.ctex. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Failed loading resource: res://addons/FMOD/editor/images/c_parameter_icon.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Error loading resource: 'res://addons/FMOD/editor/images/c_parameter_icon.svg'.
   at: load (core/core_bind.cpp:73)
WARNING: [FMOD] Failed to get icon: 0
     at: push_warning (./core/variant/variant_utility.cpp:925)
WARNING: [FMOD] Failed to get icon: 4
     at: push_warning (./core/variant/variant_utility.cpp:925)
WARNING: [FMOD] Failed to get icon: 3
     at: push_warning (./core/variant/variant_utility.cpp:925)
WARNING: [FMOD] Failed to get icon: 5
     at: push_warning (./core/variant/variant_utility.cpp:925)
WARNING: [FMOD] Failed to get icon: 8
     at: push_warning (./core/variant/variant_utility.cpp:925)
WARNING: [FMOD] Failed to get icon: 6
     at: push_warning (./core/variant/variant_utility.cpp:925)
WARNING: [FMOD] Failed to get icon: 7
     at: push_warning (./core/variant/variant_utility.cpp:925)
WARNING: [FMOD] Failed to get icon: 0
[FMOD] Cache created in res://addons/FMOD/editor/cache/fmod_project_cache.tres
     at: push_warning (./core/variant/variant_utility.cpp:925)
reimport: begin: (Re)Importing Assets steps: 13
ERROR: Unable to open file: res://.godot/imported/fmod_emitter.png-e0c584c022923e704103a5722938e616.s3tc.ctex.
	reimport: step 0: bank_icon.svg
   at: _load_data (scene/resources/texture.cpp:847)
	reimport: step 1: bus_icon.svg
ERROR: Failed loading resource: res://.godot/imported/fmod_emitter.png-e0c584c022923e704103a5722938e616.s3tc.ctex. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Failed loading resource: res://addons/FMOD/editor/images/fmod_emitter.png. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:273)
ERROR: Error loading resource: 'res://addons/FMOD/editor/images/fmod_emitter.png'.
   at: load (core/core_bind.cpp:73)
	reimport: step 2: c_parameter_icon.svg
etcpak: Encoding image size 64x64 to format DXT5 RGBA8, with mipmaps.
etcpak: Encoding took 0 ms.
	reimport: step 6: fmod_icon.svg
	reimport: step 7: folder_closed.svg
	reimport: step 8: folder_opened.svg
	reimport: step 9: l_parameter_icon.svg
	reimport: step 10: snapshot_icon.svg
	reimport: step 11: vca_icon.svg
	reimport: step 12: icon.svg
ERROR: Prepare Template: The given export path doesn't exist.
reimport: end
   at: add_message (editor/export/editor_export_platform.h:178)
ERROR: Project export for preset "Linux/X11" failed.
   at: _fs_changed (editor/editor_node.cpp:1003)
Found Android plugin FmodAndroidPlugin
EditorSettings: Save OK!
XR: Clearing primary interface
ERROR: Attempt to remove unregistered singleton: FMODStudioModule
   at: unregister_singleton (core/core_bind.cpp:1641)

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.1.2.stable.official (399c9dc393f6f84c0b4e4d4117906c70c048ecf2)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
ERROR: FATAL: Index p_index = 1 is out of bounds (size() = 0).
   at: get (./core/templates/cowdata.h:155)

================================================================
handle_crash: Program crashed with signal 4
Engine version: Godot Engine v4.1.2.stable.official (399c9dc393f6f84c0b4e4d4117906c70c048ecf2)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
ERROR: FATAL: Index p_index = 1 is out of bounds (size() = 0).
   at: get (./core/templates/cowdata.h:155)
Illegal instruction (core dumped)
Error: Process completed with exit code 132.
```

</details>

### Steps to reproduce

1. Fork https://github.com/GabLeRoux/godot-fmod-ci-example
2. Enable running actions
3. Push empty commits on both `main` and `fmod` branches using `git commit -m "empty commit to trigger pipeline" --allow-empty`
4. :fire

### Minimal reproduction project

https://github.com/GabLeRoux/godot-fmod-ci-example

**Comment by GabLeRoux** [comment]
I think there's still a few things to fix on the `main` branch because I see some problems about `export` templates, but I think the FMOD problem will still happen when these export problems get fixed.

**Comment by nospoone** [comment]
@alessandrofama I realize this isn't strictly related to your addon (which is incredible by the way! thank you for it!) but I figured that I'd post here anyways since this is what started my investigation. It still does revolve around addons and importing, though!

---

Hello! I've decided to dive deep to try an understand why this happens (as it also seems to happen for other people on other addons as well.) I've figured out the issue and potential workarounds, as we will explore together below. 
### tl;dr
There's an engine "bug" that prevents Godot from properly importing resources when it first runs, but _only_ in headless mode. It can be worked around, but other issues arise. The issue present above is actually two issues in one!
### investigation
As you already know, Godot needs to import files whenever you add things to the project. In this case, it's the FMOD addon. When it does, it creates metadata in the `.godot/imported` folder, which Godot uses to figure out what these files are and what they do.

However, when we push our project to CI, this folder *does not exist*! This is because [the official documentation](https://docs.godotengine.org/en/stable/tutorials/best_practices/version_control_systems.html#files-to-exclude-from-vcs) is clear about the fact that we need to omit the `.godot` folder. In theory, this shouldn't be an issue because when the project is first opened (for example, on a fresh clone) the editor will run through the unimported files and import them.

For some odd reason, it seems the `--headless -quit` (or a permutation of these) version of the editor exits early and isn't able to properly import things ([related but different issue](https://github.com/godotengine/godot/issues/73782).) The real fix for this is implementing an `--import` command, which has a proposal [here](https://github.com/godotengine/godot-proposals/issues/1362). There were two attempts at making this work:
- https://github.com/godotengine/godot/pull/78806
- https://github.com/godotengine/godot/pull/68461

While none of these provide a clear resolution, we can start to see a path to a solution. I'd love to try and tackle this myself, but at only week 4 of using the engine (and my cpp skills dating back to the early 00s) I'm not sure I'll be able to figure it out alone.

The main workaround, however, is just having the editor launch one time without exporting the project, closing it, and then exporting. This works on my medium-size project **without this plugin enabled**, but with another non-GDExtension plugin. As soon as I add the FMOD plugin, the project stops building and the editor throws this: `handle_crash: Program crashed with signal 11`.

While the workaround gets the import part working, it seems that the editor has further problems with loading addons. Looking at my project's log right after the reimport reveals this:

<details><summary>Relevant Log Lines</summary>

```
   ERROR: Script does not inherit from Node: res://addons/dialogue_manager/dialogue_manager.gd.
     at: _create_autoload (editor/editor_autoload_settings.cpp:423)
  ERROR: Condition "!info->node" is true. Continuing.
     at: update_autoload (editor/editor_autoload_settings.cpp:551)
  ERROR: Script does not inherit from Node: res://addons/FMOD/runtime/fmod_runtime_manager.gd.
     at: _create_autoload (editor/editor_autoload_settings.cpp:423)
  ERROR: Condition "!info->node" is true. Continuing.
     at: update_autoload (editor/editor_autoload_settings.cpp:551)
  ERROR: Condition "nc == 0" is true. Returning: nullptr
     at: instantiate (scene/resources/packed_scene.cpp:92)
  SCRIPT ERROR: Invalid set index 'editor_plugin' (on base: 'null instance') with value of type 'EditorPlugin (plugin.gd)'.
```

</details>

Also, right after the project is packed (from what I understand anyways):

<details><summary>Relevant Log Lines</summary>

```
  ERROR: Condition "!named_globals.has(p_name)" is true.
     at: remove_named_global_constant (modules/gdscript/gdscript.cpp:2047)
  ERROR: Condition "!named_globals.has(p_name)" is true.
     at: remove_named_global_constant (modules/gdscript/gdscript.cpp:2047)
  ERROR: Script does not inherit from Node: res://addons/FMOD/runtime/fmod_runtime_manager.gd.
     at: _create_autoload (editor/editor_autoload_settings.cpp:423)
  ERROR: Condition "!info->node" is true. Continuing.
     at: update_autoload (editor/editor_autoload_settings.cpp:551)
  ERROR: Attempt to disconnect a nonexistent connection from 'FileSystem:FileSystemDock#573462047246'. Signal: 'files_moved', callable: 'EditorPlugin(plugin.gd)::_on_files_moved'.
     at: _disconnect (core/object/object.cpp:1355)
  ERROR: Attempt to disconnect a nonexistent connection from 'FileSystem:FileSystemDock#573462047246'. Signal: 'file_removed', callable: 'EditorPlugin(plugin.gd)::_on_file_removed'.
     at: _disconnect (core/object/object.cpp:1355)
  ERROR: Condition "!named_globals.has(p_name)" is true.
     at: remove_named_global_constant (modules/gdscript/gdscript.cpp:2047)
  WARNING: Scan thread aborted...
       at: _notification (editor/editor_file_system.cpp:1224)
  ERROR: Attempt to remove unregistered singleton: FMODStudioModule
     at: unregister_singleton (core/core_bind.cpp:1641)
  
  ================================================================
  handle_crash: Program crashed with signal 11
  Engine version: Godot Engine v4.1.1.stable.official (bd6af8e0ea69167dd0627f3bd54f9105bda0f8b5)
  Dumping the backtrace. Please include this when reporting the bug to the project developer.
  [1] /lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x7fb84ac42520] (??:0)
  [2] unregister_fmod_types(godot::ModuleInitializationLevel) (??:0)
  [3] godot::GDExtensionBinding::deinitialize_level(void*, GDExtensionInitializationLevel) (??:0)
  [4] /home/runner/.local/share/godot/godot_executable/Godot_v4.1.1-stable_linux.x86_64() [0x47d570c] (??:0)
  [5] /home/runner/.local/share/godot/godot_executable/Godot_v4.1.1-stable_linux.x86_64() [0xe468dd] (??:0)
  [6] /lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7fb84ac29d90] (??:0)
  [7] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80) [0x7fb84ac29e40] (??:0)
  [8] /home/runner/.local/share/godot/godot_executable/Godot_v4.1.1-stable_linux.x86_64() [0xe777ae] (??:0)
  -- END OF BACKTRACE --
  ================================================================
```

</details>

While this is still related to importing, we can see that the FMOD GDExtension is throwing here. Whether this is because of the actual extension or an editor bug related to the one above, I don't know. My next steps are investigation these two options! I will then open a bug if necessary on the godot repository.

### reimport workarounds
I found a couple possible reimport workarounds, some uglier than others:
- [Monitor the CPU usage of the `godot` process and kill it once it drops below 10%](https://github.com/godotengine/godot/pull/68461#issuecomment-1328177902)
- [Run `timeout 25s godot -e --headless`](https://github.com/godotengine/godot-proposals/issues/1362)
- [Run a dummy script in the CI process to let it import (similar and more convoluted to the above, for Godot 3)](https://github.com/Kersoph/open-sequential-logic-simulation/pull/4/files)
- [Add some files from `.godot` to VCS (might have potential impacts, but is probably an easy fix)](https://github.com/dsg-jam/ld51-game/pull/38)

Funnily enough, the action you used actually has [one of the workarounds](https://github.com/abarichello/godot-ci/blob/master/Dockerfile#L58) in place **but only for non-mono builds**, it seems. However, this still fails for me because of the error mentioned above. I tried adding the timeout hack to your linked repository but couldn't make it work. I hope this can be of use to you, and I will try and implement the workarounds myself and see where this puts me.

Good luck and sorry for the essay! 🤠

**Comment by alessandrofama** [comment]
Hey, thanks for digging into this mess! This whole importing issue has been a real pain in the ass. I've even had to tell users to restart the editor after they copy the plugin files to the project. It's a terrible user experience, to be honest.

I'm wondering if those missing types/identifiers in the runtime manager are because Godot hasn't loaded the shared library yet. These types are registered in C++. What do you think?

I'm not sure what's up with those missing metadata in the icons (the error message doesn't say much), but I'm sure it'll be an easy fix once we figure out what's missing.

That crash with the unregistered singleton is bizarre, right? I haven't seen it when running the editor. You can check out the GDExtension classes being registered here: [GitHub link](https://github.com/alessandrofama/fmod-for-godot/blob/main/addons/FMOD/native/src/register_types.cpp). Maybe not deleting the FMOD studio module instance could help in your case.

In general, I think this is more of a Godot problem, and there's not much I can do except try to squash the bugs that are stopping exports. I'll dive into the singleton crash, which seems to be the biggest roadblock. If this is fixed trying the workaround suggested by @nospoone (opening and closing the editor once before exporting) should work?

**Comment by atadenizoktay** [comment]
With Godot v4.2.2, the command-line export pipeline is improved. It now includes a `--import` command to properly import everything, and it is even included by default in `--export-release` and `--export-debug` commands. More information about it can be found [here](https://godotengine.org/article/maintenance-release-godot-4-2-2-and-4-1-4/#improved-command-line-export-pipeline). Unfortunately, this still does not fix the issue mentioned above.

I would be happy to hear more about this issue. Including the `.godot` folder in the repository allows my project to successfully export a build. Still, as mentioned [here](https://docs.godotengine.org/en/stable/tutorials/best_practices/version_control_systems.html#files-to-exclude-from-vcs) the Godot documentation, it is not suggested.

**Comment by alessandrofama** [comment]
I'm closing this issue as I won't be able to work on this anymore. I'm sorry I couldn't look at this in more detail. Cheers.
